include/irr/asset/ICPUShader.h

```diff
@@ -2,185 +2,53 @@
-        }
-    };
-}//std
-
 namespace irr { namespace asset
 {
 
-struct SIntrospectionData
-{
-    struct SSpecConstant
-    {
-        enum E_TYPE
-        {
-            ET_U64,
-            ET_I64,
-            ET_U32,
-            ET_I32,
-            ET_F64,
-            ET_F32
-        };
-
-        uint32_t id;
-        size_t byteSize;
-        E_TYPE type;
-        std::string name;
-        union {
-            uint64_t u64;
-            int64_t i64;
-            uint32_t u32;
-            int32_t i32;
-            double f64;
-            float f32;
-        } defaultValue;
-    };
-    //! Sorted by `id`
-    core::vector<SSpecConstant> specConstants;
-    //! Each vector is sorted by `binding`
-    core::vector<SShaderResourceVariant> descriptorSetBindings[4];
-    //! Sorted by `location`
-    core::vector<SShaderInfoVariant> inputOutput;
-
-    struct {
-        bool present;
-        SShaderPushConstant info;
-    } pushConstant;
-
-    bool canSpecializationlesslyCreateDescSetFrom() const
-    {
-        for (const auto& descSet : descriptorSetBindings)
-        {
-            auto found = std::find_if(descSet.begin(), descSet.end(), [](const SShaderResourceVariant& bnd) { return bnd.descCountIsSpecConstant; });
-            if (found != descSet.end())
-                return false;
-        }
-        return true;
-    }
-};
-
 class ICPUShader : public IAsset
 {
 protected:
     virtual ~ICPUShader()
     {
-        if (m_spirvBytecode)
-            m_spirvBytecode->drop();
-        if (m_parsed)
-            m_parsed->drop();
-        for (auto& introData : m_introspectionCache)
-            SIntrospectionPerformer::deinitIntrospectionData(introData.second);
-        if (m_glslCompiler)
-            m_glslCompiler->drop();
+        if (m_code)
+            m_code->drop();
     }
 
 public:
-    ICPUShader(ICPUBuffer* _spirv) : m_glslCompiler(nullptr), m_spirvBytecode(_spirv), m_parsed(new IParsedShaderSource(m_spirvBytecode, core::defer)) 
-    {
-    }
-    //! While creating from GLSL source, entry point name and stage must be given in advance
-    /**
-    Note: file's name (_glsl param) will be used for resolving relative #include directives! See IGLSLCompiler::resolveIncludeDirectives() for more info.
-    */
-    ICPUShader(IGLSLCompiler* _glslcompiler, core::smart_refctd_ptr<io::IReadFile> _glsl, const std::string& _entryPoint, E_SHADER_STAGE _stage);
-    //! While creating from GLSL source, entry point name and stage must be given in advance
-    /**
-    @param _sourceName will be used for resolving relative #include directives! See IGLSLCompiler::resolveIncludeDirectives() for more info.
-    */
-    ICPUShader(IGLSLCompiler* _glslcompiler, const char* _glsl, const char* _sourceName, const std::string& _entryPoint, E_SHADER_STAGE _stage);
+    ICPUShader(ICPUBuffer* _spirv);
+    ICPUShader(const char* _glsl, const std::string& _entryPoint, E_SHADER_STAGE _stage);
```

Comment:
why are you taking the entry point here? and stage?

Reply:
hm, maybe i dont need to do this, i'll check

Reply:
you're gonna have that stuff in ISpecializationInfo/Data anyway right?

Reply:
yea